services/workers/src/test/java/org/sagebionetworks/file/worker/FileEventRecordWorkerIntegrationTest.java

```diff
@@ -0,0 +1,193 @@
+
+        //cal under test - wait for worker to send the data into s3
+        TimeUtils.waitFor(WORKER_TIMEOUT, 10_000L, () -> {
+            boolean found = getRecord(file.getDataFileHandleId(), fileDownloadRecordKey, startAfterKey);
+            return new Pair<>(found, null);
+        });
+    }
+
+    @Test
+    public void testFileUploadRecords() throws Exception {
+        //create the test data
+        createTestData(adminUserInfo);
+
+        //send data into topic FILE_EVENT
+        transactionalMessenger.publishMessageAfterCommit(new FileEvent().setObjectType(ObjectType.FILE_EVENT)
+                .setObjectId(file.getDataFileHandleId()).setTimestamp(Date.from(Instant.now()))
+                .setUserId(adminUserInfo.getId()).setFileEventType(FileEventType.FILE_UPLOAD)
+                .setFileHandleId(file.getDataFileHandleId()).setAssociateId(file.getId())
+                .setAssociateType(FileHandleAssociateType.FileEntity));
+
+        //cal under test - wait for worker to send the data into s3
+        TimeUtils.waitFor(WORKER_TIMEOUT, 10_000L, () -> {
+            boolean found = getRecord(file.getDataFileHandleId(), fileUploadRecordKey, startAfterKey);
+            return new Pair<>(found, null);
+        });
+    }
+
+    public void createTestData(final UserInfo createdByUserInfo) {
+        try {
+            project = entityManager.getEntity(createdByUserInfo, createProject(createdByUserInfo), Project.class);
+            entitiesToDelete.add(project.getId());
+
+            final S3FileHandle fileHandle;
+
+            fileHandle = fileHandleManager.createFileFromByteArray(createdByUserInfo.getId().toString(), new Date(),
+                    "Test file content".getBytes(StandardCharsets.UTF_8), "TestFile.txt", ContentTypeUtil.TEXT_PLAIN_UTF8, null);
+
+            fileHandlesToDelete.add(fileHandle);
+            file = entityManager.getEntity(createdByUserInfo, entityManager.createEntity(createdByUserInfo,
+                    new FileEntity().setName("TestFile").setParentId(project.getId()).setDataFileHandleId(fileHandle.getId()), null), FileEntity.class);
+            entitiesToDelete.add(file.getId());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String createProject(final UserInfo createdByUser) {
+        return entityManager.createEntity(createdByUser, new Project().setName(null), null);
+    }
+
+    private boolean getRecord(String fileHandleId, String key, String startAfterKey) throws IOException, JSONObjectAdapterException {
+        ListObjectsV2Result objectListing = s3Client.listObjectsV2(new ListObjectsV2Request().withBucketName(BUCKET_NAME)
+                .withPrefix(key).withStartAfter(startAfterKey));
+        for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+            S3Object object = s3Client.getObject(BUCKET_NAME, objectSummary.getKey());
```

Comment:
Make sure you put this call into a try with resources:

try (S3Object object = s3Client.getObject(BUCKET_NAME, objectSummary.getKey())) {
...
}